src/Assets/Asset.php

```diff
@@ -1696,27 +1699,190 @@ public function set_as_registered() {
 		return $this;
 	}
 
+	/**
+	 * Prints the asset using wp_print_scripts or wp_print_styles.
+	 *
+	 * @since 1.4.4
+	 *
+	 * @return static
+	 */
+	public function print_asset() {
+		$method = 'wp_print_' . $this->get_script_or_style() . 's';
+		$method( $this->get_slug() );
+		return $this;
+	}
+
+	/**
+	 * Enqueues the asset using wp_enqueue_script or wp_enqueue_style.
+	 *
+	 * @since 1.4.4
+	 *
+	 * @return static
+	 */
+	public function enqueue_asset() {
+		$method = 'wp_enqueue_' . $this->get_script_or_style();
+		$method( $this->get_slug() );
+		return $this;
+	}
+
+	/**
+	 * Registers the asset using wp_register_script or wp_register_style.
+	 *
+	 * @since 1.4.4
+	 *
+	 * @param string      $url                The URL to the asset.
+	 * @param array       $dependencies       The dependencies for the asset.
+	 * @param string      $version            The version of the asset.
+	 * @param bool|string $in_footer_or_media Whether to enqueue in the footer or the media type for the asset.
+	 *
+	 * @return static
+	 */
+	public function register_asset( string $url = '', array $dependencies = [], string $version = null, $in_footer_or_media = 'all' ) {
+		$url = $url ? $url : $this->get_url();
+		$dependencies = $dependencies ? $dependencies : $this->get_dependencies();
+		$version = $version ?? $this->get_version();
+		$in_footer_or_media = $in_footer_or_media ? $in_footer_or_media : ( $this->is_js() ? $this->is_in_footer() : $this->get_media() );
+
+		$method = 'wp_register_' . $this->get_script_or_style();
+		$method( $this->get_slug(), $url, $dependencies, $version, $in_footer_or_media );
+		return $this;
+	}
```

Comment:
This feels redundant with `register()`. If we really need this logic in here, is it best to replace the functionality of `register()`? To skip around that method sorta breaks the library's purpose.